flink-core/src/main/java/org/apache/flink/util/TimeInterval.java

```diff
@@ -0,0 +1,80 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.util;
+
+import org.apache.flink.annotation.PublicEvolving;
+
+import java.io.Serializable;
+import java.util.concurrent.TimeUnit;
+
+/**
+ * Value object that represents time interval
+ * */
+@PublicEvolving
+public final class TimeInterval implements Serializable {
```

Comment:
I like your solution to have a `TimeInterval` class. However, we already have a nearly identical class `Time` in Flink. It is currently contained in the flink-streaming-java module. Thus, in order to not replicate code, I think it would be better to move the `Time` class to flink-core and use it instead of `TimeInterval`. We can move the `org.apache.flink.streaming.api.windowing.time` package to `org.apache.flink.api.common.time`. But then we have to make sure that we update possible references in the documentation to the new location.